### PR TITLE
Video-14-Handle-Add-To-Cart-Button

### PR DESCRIPTION
**Pull request by basir**

### CODE REVIEW

File: frontend/src/screens/ProductScreen.js [hunk 1]
```diff
@@ -9,13 +9,16 @@ import Rating from '../components/Rating';
...
+  const [qty, setQty] = useState(1);
   const productDetails = useSelector((state) => state.productDetails);
   const { loading, error, product } = productDetails;
 
   useEffect(() => {
     dispatch(detailsProduct(productId));
   }, [dispatch, productId]);
-
+  const addToCartHandler = () => {
+    props.history.push(`/cart/${productId}?qty=${qty}`);
```
Comment (ashiqdammam) [inline frontend/src/screens/ProductScreen.js:20]: const navigate=useNavigate();
navigate(`/cart/${productId}?qty=${qty}`);

File: frontend/src/App.js [hunk 2]
```diff
@@ -19,6 +20,7 @@ function App() {
           </div>
         </header>
         <main>
+          <Route path="/cart/:id?" component={CartScreen}></Route>
```
Comment (ashiqdammam) [inline frontend/src/App.js:23]: <Route path="/cart/:id" element={<CartScreen />} />

File: frontend/src/screens/CartScreen.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+
+export default function CartScreen(props) {
+  const productId = props.match.params.id;
+  const qty = props.location.search
+    ? Number(props.location.search.split('=')[1])
+    : 1;
```
Comment (ashiqdammam) [inline frontend/src/screens/CartScreen.js:7]: const params = useParams();
    const productId = params.id;
    const {search} =useLocation();
    const qtyInUrl = new URLSearchParams(search).get('qty');
    const qty = qtyInUrl?Number(qtyInUrl):1;

**Review by yogesh9987 — COMMENTED** [review]
const params = useParams();
const productId = params.id;
const {search} =useLocation();
const qtyInUrl = new URLSearchParams(search).get('qty');
const qty = qtyInUrl?Number(qtyInUrl):1;

File: frontend/src/screens/ProductScreen.js (hunk 1 above)
Comment (sentinel-sam) [reply frontend/src/screens/ProductScreen.js:20]: lol, github code formatting used the backticks meant for the template string

File: frontend/src/App.js (hunk 2 above)
Comment (fanxwsam) [reply frontend/src/App.js:23]: <Route path="/cart/:id" element={<CartScreen/>} />

**Pull request merged**